File: app/src/main/java/com/example/purepawapp/ui/profile/PolicyFragment.java
package com.example.purepawapp.ui.profile;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.navigation.fragment.NavHostFragment;

import com.example.purepawapp.databinding.FragmentPolicyBinding;
import com.example.purepawapp.ui.common.BaseFragment;

public class PolicyFragment extends BaseFragment<FragmentPolicyBinding> {

    public PolicyFragment() {
        super(FragmentPolicyBinding::inflate);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        getBinding().btnBack.setOnClickListener(v -> NavHostFragment.findNavController(this).popBackStack());

        getBinding().rowPolicy1.setOnClickListener(v -> toggle(getBinding().contentPolicy1, getBinding().tvChevron1));
        getBinding().rowPolicy2.setOnClickListener(v -> toggle(getBinding().contentPolicy2, getBinding().tvChevron2));
    }

    private void toggle(View content, TextView chevron) {
        boolean expanded = content.getVisibility() == View.VISIBLE;
        content.setVisibility(expanded ? View.GONE : View.VISIBLE);
        chevron.setText(expanded ? "▼" : "▲");
    }
}

File: app/src/main/java/com/example/purepawapp/ui/profile/PolicyFragment.kt
package com.example.purepawapp.ui.profile

import android.os.Bundle
import android.view.View
import androidx.navigation.fragment.findNavController
import com.example.purepawapp.databinding.FragmentPolicyBinding
import com.example.purepawapp.ui.common.BaseFragment

class PolicyFragment : BaseFragment<FragmentPolicyBinding>(FragmentPolicyBinding::inflate) {

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.btnBack.setOnClickListener {
            findNavController().popBackStack()
        }

        binding.rowPolicy1.setOnClickListener {
            toggle(binding.contentPolicy1, binding.tvChevron1)
        }
        binding.rowPolicy2.setOnClickListener {
            toggle(binding.contentPolicy2, binding.tvChevron2)
        }
    }

    private fun toggle(content: View, chevron: android.widget.TextView) {
        val expanded = content.visibility == View.VISIBLE
        content.visibility = if (expanded) View.GONE else View.VISIBLE
        chevron.text = if (expanded) "▼" else "▲"
    }
}
